### PR TITLE
vmbackup: update AWS SDK to v2

### PR DESCRIPTION
**Pull request by zekker6**

### CODE REVIEW

File: lib/backup/s3remote/s3.go [hunk 1]
```diff
@@ -302,10 +305,10 @@ func (fs *FS) HasFile(filePath string) (bool, error) {
 		Bucket: aws.String(fs.Bucket),
 		Key:    aws.String(path),
 	}
-	o, err := fs.s3.GetObject(input)
+	o, err := fs.s3.GetObject(context.Background(), input)
 	if err != nil {
-		var ae awserr.Error
-		if errors.As(err, &ae) && ae.Code() == s3.ErrCodeNoSuchKey {
+		var nsk *types.NoSuchKey
+		if errors.As(err, &nsk) {
```
Comment (valyala) [inline lib/backup/s3remote/s3.go:311]: Probably, it would be better using `errors.Is()` here, since `nsk` isn't used here.

**Review by valyala — CHANGES_REQUESTED** [review]

File: lib/backup/s3remote/s3.go [hunk 2]
```diff
@@ -60,46 +61,46 @@ func (fs *FS) Init() error {
...
+			region, err := manager.GetBucketRegion(context.Background(), s3.NewFromConfig(cfg), fs.Bucket)
+			if err != nil {
+				outerErr = fmt.Errorf("cannot determine region for bucket %q: %w", fs.Bucket, err)
+				return
+			}
 
-	if len(fs.CustomEndpoint) > 0 {
-		// Use provided custom endpoint for S3
-		logger.Infof("Using provided custom S3 endpoint: %q", fs.CustomEndpoint)
-		// hack for https://github.com/VictoriaMetrics/VictoriaMetrics/issues/1449
```
[the remark this answers is not in the document]
Comment (zekker6) [reply lib/backup/s3remote/s3.go:81]: I've tested this on self-hosted Minio server with `AWS_REGION` not being set and set to empty string, SDK uses region to `us-west-1` by default.

**Review by valyala — APPROVED** [review]

**Comment by valyala** [comment]
@zekker6 , thanks for the upgrade of AWS SDK from v1 to v2!

**Comment by valyala** [comment]
@zekker6 , could you address the following errors in the enterprise repository of VictoriaMetrics?

```
lib/backup/s3remote/s3.go:290:13: undefined: s3manager
lib/backup/s3remote/s3.go:290:66: undefined: aws.BackgroundContext
lib/backup/s3remote/s3.go:290:87: undefined: s3manager
lib/backup/s3remote/s3.go:356:15: fs.s3.ListObjectsV2Pages undefined (type *s3.Client has no field or method ListObjectsV2Pages)
```

File: lib/backup/s3remote/s3.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (valyala) [reply lib/backup/s3remote/s3.go:81]: FYI, the related pull request - https://github.com/VictoriaMetrics/VictoriaMetrics/pull/3224

**Pull request merged**